crypto/src/main/java/org/springframework/security/crypto/bcrypt/BCrypt.java

```diff
@@ -912,17 +914,6 @@ public static boolean checkpw(String plaintext, String hashed) {
 	}
 
 	static boolean equalsNoEarlyReturn(String a, String b) {
-		char[] caa = a.toCharArray();
-		char[] cab = b.toCharArray();
-
-		if (caa.length != cab.length) {
-			return false;
-		}
-
-		byte ret = 0;
-		for (int i = 0; i < caa.length; i++) {
-			ret |= caa[i] ^ cab[i];
-		}
-		return ret == 0;
+		return MessageDigest.isEqual(a.getBytes(StandardCharsets.UTF_8), b.getBytes(StandardCharsets.UTF_8));
```

Comment:
@jzheaux Is there a functional difference between `java.lang.String#getBytes()` and `org.springframework.security.crypto.codec.Utf8#encode()`?

I've used the first here, but I've also seen the latter being used at other places in the code.